src/modules/featureform/featureform-directive.js

```diff
@@ -21,31 +21,31 @@ angular.module('anol.featureform')
      * @name anol.featureform.directive:anolFeatureForm
      *
      * @restrict A
-     * @requires pascalprecht.$translate
      *
      * @param {string} templateUrl Url to template to use instead of default one
      * @param {ol.Feature} feature Feature to show properties for
      * @param {anol.layer.Feature} layer Layer of feature
-     * @param {FormField[]} formFields The options for each form field
-     * @param {Object<string, string>} formValues The values for each form field
-     * @param {string} translationNamespace Namespace to use in translation table. Default "featureform".
+     * @param {FormField[]} pointFields The form fields for points
+     * @param {FormField[]} lineFields The form fields for lines
+     * @param {FormField[]} polygonFields The form fields for polygons
+     * @param {boolean} [highlightInvalid] show if fields are invalid
      *
      * @description
      * Creates a form to edit defined feature properties.
      *
      */
-    .directive('anolFeatureForm', ['$templateRequest', '$compile', '$translate',
-        function($templateRequest, $compile, $translate) {
+    .directive('anolFeatureForm', ['$templateRequest', '$compile',
+        function($templateRequest, $compile) {
             return {
                 restrict: 'A',
                 require: '?^anolFeaturePopup',
                 scope: {
                     'feature': '=',
                     'layer': '=', // TODO: is this needed?
-                    'selects': '=', // TODO: is this needed?
-                    'formFields': '=',
-                    'formValues': '=',
-                    'translationNamespace': '@' // TODO: is this needed?
+                    'pointFields': '=?',
+                    'lineFields': '=?',
+                    'polygonFields': '=?',
+                    'highlightInvalid': '=?'
```

Comment:
We can probably set the type of `highlightInvalid` to `'@?'` or `'<?'`, as we do not need a two-way data-binding here. See also https://code.angularjs.org/1.7.9/docs/api/ng/service/$compile#-scope-